sympy/solvers/decompogen.py

```diff
@@ -51,6 +54,21 @@ def decompogen(f, symbol):
         result += [f.subs(arg, symbol)] + decompogen(arg, symbol)
         return result
 
+    # ===== Min/Max Functions ===== #
+    if isinstance(f, (Min, Max)):
+        if And(*[a.has(symbol) for a in f.args]):
+            raise TypeError('cannot decompose %s' % f)
+        args = list(f.args)
+        iargs = [i for i, a in enumerate(f.args) if a.has(symbol)]
+        if not iargs:
+            return result
```

Comment:
should this be `result + [f]`?  And maybe the first check can be rolled into the second:
```python
if len(iargs) == len(args):
    raise TypeError('cannot decompose %s' % f)
```
The case of `not iargs` can be omited since the test at the outset affirms that `f.has(symbol)`.

Reply:
Yeah, the code is now much more simplified.